launch/launch/launcher.py

```diff
@@ -51,13 +57,35 @@ def add_launch_descriptor(self, launch_descriptor):
 
             self.task_descriptors.append(task_descriptor)
 
+    def interrupt_launch(self):
+        with self.loop_lock:
+            if self.loop is not None:
+                self.loop.call_soon_threadsafe(self.interrupt_launch_non_threadsafe)
+
+    def interrupt_launch_non_threadsafe(self):
+        self.interrupt_future.set_result(True)
+
+    def wait_on_launch_to_finish(self, timeout):
+        return self.launch_complete.wait(timeout)
+
+    def launch_is_running(self):
```

Comment:
This is on the `DefaultLauncher` class which is not a Thread. Presumably you can call launch multiple times on this object, so I think it isn't even comparable to a thread.

The `AsynchronousLauncher` class, which inherits from `threading.Thread`, only has the custom `terminate` function and otherwise the `Thread.wait` and `Thread.is_alive` methods are used.